hawkbit-rest/hawkbit-mgmt-resource/src/main/java/org/eclipse/hawkbit/mgmt/rest/resource/MgmtSoftwareModuleMapper.java

```diff
@@ -155,6 +161,17 @@ static void addLinks(final Artifact artifact, final MgmtArtifact response) {
                         .expand());
     }
 
+    static void addLinks(final Artifact artifact, final MgmtArtifact response,
+            final ArtifactUrlHandler artifactUrlHandler, final SystemManagement systemManagement) {
+
+        List<ArtifactUrl> urls = artifactUrlHandler.getUrls(
+                new URLPlaceholder(systemManagement.getTenantMetadata().getTenant(),
+                        systemManagement.getTenantMetadata().getId(), null, null,
+                        new URLPlaceholder.SoftwareData(artifact.getSoftwareModule().getId(), artifact.getFilename(),
+                                artifact.getId(), artifact.getSha1Hash())), ApiType.MGMT, null);
+        response.add(Link.of(urls.get(0).getRef()).withRel("download").expand());
```

Comment:
I think we shouldn't hardcode to use the first entry of the list with a fixed rel value. Instead we should do something like:
```suggestion
        urls.forEach(entry -> response.add(Link.of(entry.getRef()).withRel(entry.getRel()).expand()));
```